front/pages/w/[wId]/assistant/new.tsx

```diff
@@ -222,235 +211,190 @@ export default function AssistantNew({
     setShowQuickGuide(false);
   });
 
+  useEffect(() => {
+    if (!quickGuideSeen && !isQuickGuideSeenError && !isQuickGuideSeenLoading) {
+      // Quick guide has never been shown, lets show it.
+      setShowQuickGuide(true);
+    }
+  }, [isQuickGuideSeenError, isQuickGuideSeenLoading, quickGuideSeen]);
+
+  useEffect(() => {
+    setGreeting(getRandomGreetingForName(user.firstName));
+  }, [user]);
+
+  const handleAssistantClick = useCallback(
+    async (agent: LightAgentConfigurationType) => {
+      // scroll to inputbar title
+      const scrollContainerElement = document.getElementById(
+        "assistant-input-header"
+      );
+      console.log(
+        "y delta of scroll container with current scroll position:" +
+          scrollContainerElement?.getBoundingClientRect().top
+      );
+
+      if (scrollContainerElement) {
+        scrollContainerElement.scrollIntoView({ behavior: "smooth" });
+        const waitScrollDelay =
+          -scrollContainerElement?.getBoundingClientRect().top * 0.8;
+        // wait for end of scroll
+        await new Promise((resolve) => setTimeout(resolve, waitScrollDelay));
+      }
+
+      setSelectedAssistant({
+        configurationId: agent.sId,
+      });
+
+      // animate input bar
+      setAnimate(true);
+      await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
`nit` create a helper `setTimeoutAsync`.

Reply:
I reaaly think `await sleep(ms)` is better than `await setTimeoutAsync(() => {}, ms)` :upside_down_face: